dtfs-central-api/src/v1/controllers/utilisation-report-service/put-utilisation-report-status.controller.ts

```diff
@@ -0,0 +1,97 @@
+
+type RequestBody = {
+  user: TfmUser;
+  reportsWithStatus: ReportWithStatus[];
+};
+
+class PayloadError extends Error {
+  status: number;
+
+  constructor(message: string) {
+    super(message);
+    this.status = 400;
+  }
+}
+
+const getUpdateInstructionsWithReportDetails = (
+  status: UtilisationReportReconciliationStatus,
+  reportDetails: ReportDetails,
+): UpdateUtilisationReportStatusInstructions => {
+  const filter: ReportFilter = {
+    month: reportDetails.month,
+    year: reportDetails.year,
+    'bank.id': reportDetails.bankId,
+  };
+  return { status, filter };
+};
+
+const getUpdateInstructionsWithReportId = (status: UtilisationReportReconciliationStatus, reportId: ReportId): UpdateUtilisationReportStatusInstructions => {
+  const filter: ReportFilter = {
+    _id: new ObjectId(reportId.id),
+  };
+  return { status, filter };
+};
+
+const getUpdateInstructions = (reportWithStatus: ReportWithStatus): UpdateUtilisationReportStatusInstructions => {
+  const { status, report } = reportWithStatus;
+
+  const reportContainsReportId = 'id' in report;
+  const reportContainsReportDetails = 'bankId' in report && 'month' in report && 'year' in report;
+  if (!reportContainsReportId && !reportContainsReportDetails) {
+    throw new PayloadError("Request body item 'reportsWithStatus' supplied does not match required format");
+  }
+
+  if (reportContainsReportDetails) {
+    return getUpdateInstructionsWithReportDetails(status, report);
+  }
+  return getUpdateInstructionsWithReportId(status, report);
+};
+
+export const putUtilisationReportStatus = async (req: Request<{}, {}, RequestBody>, res: Response) => {
+  try {
+    const { reportsWithStatus, user } = req.body;
+
+    if (!user || !user._id || !user.firstName || !user.lastName) {
+      throw new PayloadError("Request body item 'user' supplied does not match required format");
+    }
+    const uploadedByUserDetails: UploadedByUserDetails = {
+      id: user._id.toString(),
+      firstname: user.firstName,
+      surname: user.lastName,
+    };
+
+    if (!reportsWithStatus) {
+      throw new PayloadError("Request body item 'reportsWithStatus' supplied does not match required format");
+    }
+    const updateInstructions = reportsWithStatus.map((reportWithStatus) => getUpdateInstructions(reportWithStatus));
+
+    const client: MongoClient = await db.getClient();
+    const session = client.startSession();
+    await session.withTransaction(async () => {
+      await updateManyUtilisationReportStatuses(updateInstructions, uploadedByUserDetails);
+    });
+    await session.endSession();
+
+    return res.sendStatus(200);
+  } catch (error: any) {
+    console.error('Error updating utilisation report status:', error);
+    if (error instanceof PayloadError) {
+      return res.status(error.status).send({ error: 'Update utilisation report status request failed' });
```

Comment:
As we now know this is a `PayloadError` I think it's safe to return the `error.message` in the request. e.g.
```
return res.status(error.status).send({ error: error.message });
```